journald/journald.cpp

```diff
@@ -293,6 +298,22 @@ int main(int argc, char** argv)
     LOG(WARNING) << warning.message;
   }
 
+  // Change the current working directory to the container's sandbox directory.
+  // This is to handle the case that the nested container's user is a non-root
+  // user and different from its parent container's user, in which case the
+  // nested container's user (i.e.,`flags.user`) has no permission to access
+  // `flags.logrotate_filename` since it has no permission to traverse its
+  // parent container's sandbox directory whose permissions is 0750.
+  if (flags.logrotate_filename.isSome()) {
```

Comment:
This is a required option (checked with a flag validation lambda), so I would prefer:
```
CHECK_SOME(flags.logrotate_filename);

Try<Nothing> result = os::chdir(Path(flags.logrotate_filename.get()).dirname());
...
```

Reply:
@kaysoky That's a bit confused to me, if `flags.destination_type` is `journald` rather than `logrotate`, do we still need `flags.logrotate_filename` to be specified? It seems making `flags.logrotate_filename` required in the case of `journald` does not make sense since we do not need to touch that file at all.

Reply:
For the most part, we can assume that the journald option is never used.  The performance of journald was terrible for task logging and so we default to logrotate instead.

Reply:
Got it, and I have updated the patch accordingly.

I do not have write access, could you please help do the merge?